Uchu.World/Objects/Components/ReplicaComponents/InventoryComponent.cs

```diff
@@ -229,13 +233,57 @@ private async Task EnsureItemUnEquipped(Item item)
         {
             if (GameObject.TryGetComponent<SkillComponent>(out var skillComponent))
                 await skillComponent.UnequipItemAsync(item);
-            
+
             await OnUnEquipped.InvokeAsync(item);
-            
+
             item.IsEquipped = false;
             Items.Remove(GetSlot(item.Id));
         }
 
+        /// <summary>
+        /// Pushes the currently equipped item state to an internal buffer. Can be reverted by calling
+        /// <see cref="PopEquippedItemState"/>
+        /// </summary>
+
+        public void PushEquippedItemState()
+        {
+            EquippedItemStack.Clear();
+            foreach (var pair in Items)
+            {
+                Logger.Debug($"Pushing {pair.Key}: {pair.Value}");
+                EquippedItemStack.Add(pair.Key, pair.Value);
+            }
+        }
+
+        /// <summary>
+        /// Restores the state saved by <see cref="PushEquippedItemState"/>
+        /// </summary>
+        public async void PopEquippedItemState()
+        {
+            Logger.Debug("PopEquippedItemState");
+            if (EquippedItemStack == null || EquippedItemStack.Count == 0)
+                return;
+
+            Logger.Debug($"Equipped Items: {EquippedItems.Length}");
+            Logger.Debug($"Items on stack: {EquippedItemStack.Count}");
+
+            foreach (var item in EquippedItems)
+            {
+                Logger.Debug($"Unequip {item}");
+                await UnEquipItemAsync(item, false);
```

Comment:
Why does serialize need to be false here? Is it just to avoid sending many messages in rapid succession?

Reply:
Yes that's the biggest reason for it. I also wanted to make sure that the client would not get any messages in the wrong order (not sure if this is necessary).